src/hooks/use-load-image.js

```diff
@@ -0,0 +1,33 @@
+import { usePromise } from "@nebula.js/stardust";
+
+// Create a promise that ensures the button background image is loaded properly
+const imagePromise = (url) =>
+  new Promise((resolve, reject) => {
+    const img = new Image();
+    img.src = url;
+    img.onload = () => resolve(img);
+    img.onerror = () => reject(img);
+  });
+
+const useLoadImage = (layout) => {
+  const { url } = layout.style.background;
+  const [isBackgroundLoaded = false, error] = usePromise(async () => {
+    if (url && url.qStaticContentUrl && url.qStaticContentUrl.qUrl) {
+      try {
+        await imagePromise(url.qStaticContentUrl.qUrl);
+        return true;
+      } catch {
+        return false;
+      }
+    }
+    return false;
+  }, [layout]);
+  if (error) {
```

Comment:
you are catching the possible error from `imagePromise`, I don't think you get one from `usePromise`.

Reply:
Do you mean returning `false` ? fixed it.

Reply:
No, I mean that you are basically handling the error twice basically. One try catch block inside the usePromise, and then checking the error outside usePromise as well. Not sure, but I think that usePromise will never return an error since you catch it in usePromise

Reply:
Ah, true! will remove it.